src/broadcasting.jl

```diff
@@ -1,19 +1,56 @@
-# See: https://docs.julialang.org/en/v1/manual/interfaces/#man-interfaces-broadcasting-1
+"""
+    NamedDimsStyle{S}
+This is a `BroadcastStyle` for NamedDimsArray's
+It preserves the dimension names.
+`S` should be the `BroadcastStyle` of the wrapped type.
+"""
+struct NamedDimsStyle{S <: BroadcastStyle} <: AbstractArrayStyle{Any} end
+NamedDimsStyle(::S) where {S} = NamedDimsStyle{S}()
+NamedDimsStyle(::S, ::Val{N}) where {S,N} = NamedDimsStyle(S(Val(N)))
+NamedDimsStyle(::Val{N}) where N = NamedDimsStyle{DefaultArrayStyle{N}}()
+function NamedDimsStyle(a::BroadcastStyle, b::BroadcastStyle)
+    inner_style = BroadcastStyle(a, b)
+
+    # if the inner_style is Unknown then so is the outer-style
+    if inner_style isa Unknown
+        return Unknown()
+    else
+        return NamedDimsStyle(inner_style)
+    end
+end
+function Base.BroadcastStyle(::Type{<:NamedDimsArray{L, T, N, A}}) where {L, T, N, A}
+    inner_style = typeof(BroadcastStyle(A))
+    return NamedDimsStyle{inner_style}()
+end
 
-Base.BroadcastStyle(::Type{<:NamedDimsArray}) = Broadcast.ArrayStyle{NamedDimsArray}()
 
-function Base.similar(
-    bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{NamedDimsArray}},
-    ::Type{T}
-) where T
+Base.BroadcastStyle(::NamedDimsStyle{A}, ::NamedDimsStyle{B}) where {A, B} = NamedDimsStyle(A(), B())
+Base.BroadcastStyle(::NamedDimsStyle{A}, b::B) where {A, B} = NamedDimsStyle(A(), b)
+Base.BroadcastStyle(a::A, ::NamedDimsStyle{B}) where {A, B} = NamedDimsStyle(a, B())
+Base.BroadcastStyle(::NamedDimsStyle{A}, b::DefaultArrayStyle) where {A} = NamedDimsStyle(A(), b)
+Base.BroadcastStyle(a::AbstractArrayStyle{M}, ::NamedDimsStyle{B}) where {B,M} = NamedDimsStyle(a, B())
+
+
+function unwrap_broadcasted(bc::Broadcasted{NamedDimsStyle{S}}) where S
+    inner_args = unwrap_broadcasted.(bc.args)
```

Comment:
Prefer map when mapping a function over a collection. 
```suggestion
    inner_args = map(unwrap_broadcasted, bc.args)
```